src/main/java/swingtree/UIForTabbedPane.java

```diff
@@ -442,6 +446,82 @@ public final UIForTabbedPane<P> add( Tab tab )
         ._this();
     }
 
+    /**
+     * Dynamically generates tabs for items in a {@link Vals} list and automatically updates them
+     * when the items change. The items are typically view model instances, but can be any type.
+     * <p>
+     * The provided {@link TabSupplier} lambda is invoked with each item from the list,
+     * returning a {@link Tab} to be added to the {@link JTabbedPane} wrapped by this builder.
+     * <p>
+     * <b>Note:</b> Binding tabs to a {@link Vals} list assumes no other tabs are present.
```

Comment:
That is a great insight and really good that you mention this assumption here! We should probably also log warnings if we encounter something that breaks this assumption!